content/post/kubestash-v2023.12.28/index.md

```diff
@@ -0,0 +1,141 @@
+      metadata:
+       name: demo-pod-hook
+       namespace: demo
+      spec:
+        usagePolicy:
+          allowedNamespaces:
+            from: All
+        params:
+        - name: TEST
+          usage: This is a test param
+          required: false
+        action:
+          exec:
+            command:
+            - /bin/sh
+            - -c
+            - echo data_test > /source/data/data.txt
+        executor:
+          type: Pod
+          pod:
+            selector: name=test-app, test=hook
+            strategy: ExecuteOnOne
+     ```
+
+3. You can now trigger backup for specific sessions using KubeStash CLI ([#10](https://github.com/kubestash/cli/pull/10)). Here is an example:
+   
+    Lets, the applied BackupConfiguration is configured with multiple sessions. Now you want to trigger backup for specific sessions. To do so, you have to provide comma seperated sessions name using `sessions` flag.
+    ```bash
+   $ kubectl kubestash trigger -n <namespace> <backupconfiguration-name> --sessions=<sessions-name>
+    ```
+   
+4. You can now apply the Pod Security Policy profile to `Restricted`. This profile is a highly restrictive policy aligned with best practices for hardening Pods. Learn more about pod security [HERE](https://kubernetes.io/docs/concepts/security/pod-security-standards/#restricted). You can restrict a namespace by the following command:
+
+   `kubectl label namespace <namespace_name> pod-security.kubernetes.io/enforce=restricted`
+
+    Use below command to install KubeStash enforcing Pod Security Policy profile to `Restricted`:
+    ```bash 
+    helm install kubestash oci://ghcr.io/appscode-charts/kubestash \
+          --version v2023.12.28 \
+          --namespace kubestash --create-namespace \
+          --set-file global.license=<LICENCE_FILE> \
+          --wait --burst-limit=10000 --debug \
+          --set kubestash-operator.operator.securityContext.seccompProfile.type=RuntimeDefault \
+          --set kubestash-operator.rbacproxy.securityContext.seccompProfile.type=RuntimeDefault \
+          --set kubestash-operator.cleaner.securityContext.seccompProfile.type=RuntimeDefault
+    ```
+   
+    To enforce Pod Security Policy profile to `Restricted` in storage initialize/cleanup jobs or in backup/restore jobs, add Pod Security and Container Security to BackupStorage, BackupConfiguration and RestoreSession as below:
+
+    Pod Security:
+    ```yaml
+    securityContext:
+      runAsUser: 65535
+      runAsNonRoot: true
+      seccompProfile:
+        type: RuntimeDefault
+    ```
+    Container Security:
+    ```yaml
+    securityContext:
+      allowPrivilegeEscalation: false
+      capabilities:
+        drop:
+        - ALL
+    ```
+
+### Improvements & bug fixes
+- Now you can create multiple `RetentionPolicy` along with one default `RetentionPolicy` in a namespace. There can be maximum one default `RetentionPolicy` in each namespace.
+- Fixed a bug in validation for selector type usage policy.
```

Comment:
- Fixed a bug that was preventing the creation of multiple Retention Policies when a default one exists in a namespace.